admin/installer/cloudformation.py

```diff
@@ -0,0 +1,293 @@
+"""
+Troposphere script to generate an AWS CloudFormation JSON template.
+
+Sample usage:
+python cloudformation.py > flocker-cluster.cloudformation.json
+
+Resulting JSON template has the following blueprint to describe the
+desired stack's resources and properties:
+1 Control Node with Flocker Control Service, (TLS-enabled) Swarm Manager,
+                    (TLS-enabled) Docker, Ubuntu 14.04
+2 Agent Nodes with Flocker Dataset Agent, Swarm Agent, (TLS-enabled) Docker,
+                   Ubuntu 14.04
+1 Client Node with Flockerctl, Docker, Docker-compose, Ubuntu 14.04
+
+To manifest the blueprint, please input the JSON template at AWS CloudFormation
+Create Stack console (after replacing ``us-east-1`` with your Region):
+https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new
+
+"""
```

Comment:
For the sake of future maintainers, could I ask for a blob with the architecture here.

Specifically:
- Brief overview of the fact that you use  User Data [http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html#instancedata-add-user-data] To inject startup scripts to the instances.
- The user data scripts configure and install software, and then makes a curl request to a <cloudformation wait condition? (is that what it's called)> one the node is ready to use.
- If any of the wait conditions are not met after 600 seconds the script times out, and cloud formation rolls back the whole thing.